compiler/rustc_middle/src/middle/resolve_lifetime.rs

```diff
@@ -15,8 +15,6 @@ use rustc_macros::HashStable;
 pub enum LifetimeDefOrigin {
```

Comment:
This enum was just used to forbid in-band lifetimes in some places. It can be fully removed now.